ipatests/test_integration/test_adtrust_install.py

```diff
@@ -266,3 +270,81 @@ def test_ipa_user_s4u2self_pac(self):
         finally:
             tasks.kinit_admin(self.master)
             self.master.run_command(['ipa', 'user-del', user])
+
+
+class TestIpaNisManage(IntegrationTest):
+    """
+    Tests for ipa-nis-manage CLI tool
+    """
+    @classmethod
+    def install(cls, mh):
+        tasks.install_master(cls.master)
+
+    def get_dirsrv_id(self):
+        serverid = realm_to_serverid(self.master.domain.realm)
+        return("dirsrv@%s.service" % serverid)
+
+    def test_ipa_nis_manage_enable(self):
+        """
+        This testcase checks if ipa-nis-manage enable
+        command enables plugin on an IPA master
+        """
+        dirsrv_service = self.get_dirsrv_id()
+        console_msg = (
+            "Enabling plugin\n"
+            "This setting will not take effect until you restart Directory Server.\n"
+            "The rpcbind service may need to be started"
+        )
+        status_msg = "Plugin is enabled"
+        tasks.kinit_admin(self.master)
+        result = self.master.run_command(
+            ["ipa-nis-manage", "enable"],
+            stdin_text=self.master.config.admin_password,
+        )
+        assert console_msg in result.stdout_text
+        self.master.run_command(["systemctl", "restart", dirsrv_service])
+        time.sleep(DIRSRV_SLEEP)
+        self.master.run_command(["systemctl", "restart", "rpcbind"])
+        result = self.master.run_command(
+            ["ipa-nis-manage", "status"],
+            stdin_text=self.master.config.admin_password,
+        )
+        assert status_msg in result.stdout_text
```

Comment:
If you want to not trust the command output you can check directly whether the plugin is enabled by pulling the entry:

DN(('cn', 'NIS Server'), ('cn', 'plugins'), ('cn', 'config'))

and ensuring that nsslapd-pluginenabled.lower() == 'on'

Similar with disable.

Reply:
Added the change to pull the entry when plugin is enabled/disabled.